Code/Tools/AssetProcessor/native/utilities/ApplicationManagerBase.cpp

```diff
@@ -434,63 +434,40 @@ void ApplicationManagerBase::DestroyPlatformConfiguration()
-
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileAdded, [](QString path) { AZ::Interface<AssetProcessor::ExcludedFolderCacheInterface>::Get()->FileAdded(path); });
-
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileAdded,
-            m_fileProcessor.get(), &AssetProcessor::FileProcessor::AssessAddedFile);
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileRemoved,
-            m_fileProcessor.get(), &AssetProcessor::FileProcessor::AssessDeletedFile);
-
-        m_folderWatches.push_back(AZStd::unique_ptr<FolderWatchCallbackEx>(newFolderWatch));
-        m_watchHandles.push_back(m_fileWatcher.AddFolderWatch(newFolderWatch));
+        m_fileWatcher.AddFolderWatch(info.ScanPath(), info.RecurseSubFolders());
     }
 
-    // also hookup monitoring for the cache (output directory)
     QDir cacheRoot;
     if (AssetUtilities::ComputeProjectCacheRoot(cacheRoot))
     {
-        FolderWatchCallbackEx* newFolderWatch = new FolderWatchCallbackEx(cacheRoot.absolutePath(), "", true);
+        m_fileWatcher.AddFolderWatch(cacheRoot.absolutePath(), true);
+    }
 
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileAdded, [this](QString path) { m_fileStateCache->AddFile(path); });
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileModified, [this](QString path) { m_fileStateCache->UpdateFile(path); });
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileRemoved, [this](QString path) { m_fileStateCache->RemoveFile(path); });
+    connect(&m_fileWatcher, &FileWatcher::fileAdded,
+        m_assetProcessorManager, &AssetProcessor::AssetProcessorManager::AssessAddedFile);
+    connect(&m_fileWatcher, &FileWatcher::fileModified,
+        m_assetProcessorManager, &AssetProcessor::AssetProcessorManager::AssessModifiedFile);
+    connect(&m_fileWatcher, &FileWatcher::fileRemoved,
+        m_assetProcessorManager, &AssetProcessor::AssetProcessorManager::AssessDeletedFile);
 
-        // we only care about cache root deletions.
-        QObject::connect(newFolderWatch, &FolderWatchCallbackEx::fileRemoved,
-            m_assetProcessorManager, &AssetProcessor::AssetProcessorManager::AssessDeletedFile);
+    connect(&m_fileWatcher, &FileWatcher::fileAdded, [this](QString path) { m_fileStateCache->AddFile(path); });
+    connect(&m_fileWatcher, &FileWatcher::fileModified, [this](QString path) { m_fileStateCache->UpdateFile(path); });
+    connect(&m_fileWatcher, &FileWatcher::fileRemoved, [this](QString path) { m_fileStateCache->RemoveFile(path); });
 
-        m_folderWatches.push_back(AZStd::unique_ptr<FolderWatchCallbackEx>(newFolderWatch));
-        m_watchHandles.push_back(m_fileWatcher.AddFolderWatch(newFolderWatch));
-    }
+    connect(newFolderWatch, &FileWatcher::fileAdded, [](QString path) { AZ::Interface<AssetProcessor::ExcludedFolderCacheInterface>::Get()->FileAdded(path); });
```

Comment:
I don't see the `newFolderWatch1` variable in this function.

Reply:
This was a bad merge, it had been added in `development`, and I forgot to fix that variable name. I've fixed it now.